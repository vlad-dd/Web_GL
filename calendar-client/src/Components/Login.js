import React from 'react';
import { Link } from 'react-router-dom';

const Login = (props) => {
    return (
        <div>
            <img class="image" src="http://www.cfdesigner.com/codepen/rocket-page-logo.png" alt="rocket-image" />
            <form className="login_form d-table mr-auto ml-auto w-500px bg-light">
                <h2>Log in</h2>
                <div className="form-group">
                    <label for="exampleInputEmail1">Email address</label>
                    <input onChange={props.updateInput} type="email" className="form-control" name="email" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email" />
                    <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small>
                </div>
                <div className="form-group">
                    <label for="exampleInputPassword1">Password</label>
                    <input onChange={props.updateInput} type="password" className="form-control" name="password" id="exampleInputPassword1" placeholder="Password" />
                </div>

                <Link to='/register' className="d-block">Don't have an account? Register here...</Link>

                <button type="submit" className="btn btn-primary" onClick={props.tryLogin}>Submit</button>
            </form>
        </div>
    );
}

export default Login;