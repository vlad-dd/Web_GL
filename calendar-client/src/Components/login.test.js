import React from 'react';
import Login from './Login'




test('the login data is correct ', () => {
    expect('Login').toMatchSnapshot();
});


test('map calls its argument with a non-null argument', () => {
    const mock = jest.fn();
    [1].map(x => mock(x));
    expect(mock).toBeCalledWith(expect.anything());
  });

  it("saves on submit", function(){
    Login.done = jest.genMockFunction();
    Login.insideDone = jest.genMockFunction();
  TestUtils.Simulate.submit(form-group);
  expect(Login.done).toBeCalled(); 
  expect(Login.insideDone).toBeCalled(); 
});

describe('InputWithValidation tests', () => {
    it('should render component with initial state', () => {
        const props = {
            handleClick: jest.fn()
        }
        const wrapper = shallow(<Login  {...props}/>);
        expect(wrapper.state('value').toBe(''))
        expect(wrapper.find('input').length).toBe(0)
        expect(wrapper.find('button').length).toBe(0)

    })
});

it('should hide warning input when text will contain @', () => {
    const props = {
        handleClick: jest.fn()
    };

    const wrapper = shallow(<Login {...props} />);
    wrapper.find('input').simulate('change', {target: {value: 'not an email'}});
    expect(wrapper.find('button').length).toBe(1);

    wrapper.find('input').simulate('change', {target: {value: 'test@test.com'}});
    expect(wrapper.find('button').length).toBe(1);

  
    
});