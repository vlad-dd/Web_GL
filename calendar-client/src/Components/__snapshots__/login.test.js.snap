// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the login data is correct  1`] = `"Login"`;
